indico/htdocs/js/indico/widgets/person_link_widget.js

```diff
@@ -346,5 +352,42 @@
             }
             $field.principalfield('refresh');
         });
+
+        $field.closest('form').on('ajaxDialog:validateBeforeSubmit', function(evt) {
+            var $this = $(this);
+            if (options.require.primaryAuthor || options.require.secondaryAuthor || options.require.submitter ||
+                options.require.speaker) {
+                var hiddenData = JSON.parse($field.val());
+                var $formField = $field.closest('.form-field');
+                var $formGroup = $field.closest('.form-group');
+                var hasError = false;
+
+                if (options.require.speaker && _.indexOf(_.pluck(hiddenData, 'isSpeaker'), true) === -1) {
+                    hasError = true;
+                    $formField.data('error', $T.gettext('You must add at least 1 speaker'));
+                }
+                if (options.require.submitter && _.indexOf(_.pluck(hiddenData, 'isSubmitter'), true) === -1) {
```

Comment:
You check all requirements but then you show only the error message corresponding to the last missing requirement. Why not make this an `if .. else if` chain? That way we can stop checking after the first one that doesn't apply.